djmoney/money.py

```diff
@@ -24,8 +24,21 @@ class Money(DefaultMoney):
 
     def __init__(self, *args, **kwargs):
         self.decimal_places = kwargs.pop("decimal_places", DECIMAL_PLACES)
+        self._decimal_places_display = kwargs.pop("decimal_places_display", None)
         super().__init__(*args, **kwargs)
 
+    @property
+    def decimal_places_display(self):
+        if self._decimal_places_display is None:
+            self._decimal_places_display = DECIMAL_PLACES_DISPLAY.get(self.currency.code, self.decimal_places)
```

Comment:
I am not sure about caching on the attribute level. What if somebody will use `override_settings(CURRENCY_DECIMAL_PLACES_DISPLAY=...)` in their test, but a test money object is set up and this property is evaluated before the decorator is applied? In this case, it might be confusing - no explicit `decimal_places_display` in the object, but the settings change won't affect it. It is a kinda special example but might cause problems for users

Reply:
I usually wouldn't change production code for the sake of test-code, but I also do not see a specific reason to keep the local caching - so I removed it :)

PS: I was actually wondering if the override settings would work without the cache as the `from settings import DECIMAL...` thing probably keeps its value?

Reply:
You are right - I overlooked it :(